python/src/trezorlib/debuglink.py

```diff
@@ -292,12 +301,42 @@ def input(
         decision = messages.DebugLinkDecision(
             button=button, swipe=swipe, input=word, x=x, y=y, wait=wait, hold_ms=hold_ms
         )
+
         ret = self._call(decision, nowait=not wait)
         if ret is not None:
             return LayoutContent(ret.lines)
 
+        # Getting the current screen after the (nowait) decision
+        self.save_current_screen_if_relevant(wait=False)
+
         return None
 
+    def save_current_screen_if_relevant(self, wait: bool = True) -> None:
+        """Optionally saving the textual screen output."""
+        if self.screen_text_file is not None:
+            if wait:
+                layout = self.wait_layout()
+            else:
+                layout = self.read_layout()
+            self.save_debug_screen(layout.lines)
+
+    def save_debug_screen(self, lines: List[str]) -> None:
+        if self.screen_text_file is not None:
+            if not self.screen_text_file.exists():
```

Comment:
I don't think this is necessary? `open("x", "a")` will create empty file if it doesn't exist, no?

moreso that you clear the file in `set_screen_text_file` anyway

Reply:
Right, tried `with open("x", "a") as f: f.write("xx")` and the file was created. Deleted in [78785b0](https://github.com/trezor/trezor-firmware/pull/2750/commits/78785b0f6d9a163e752675f37341ae75cde08280)